mindmeld/components/request.py

```diff
@@ -246,6 +246,32 @@ def nlp_params(self):
             )
         }
 
+    def to_dict(self):
+        fields = [
+            "allowed_intents",
+            "target_dialogue_state",
+            "time_zone",
+            "timestamp",
+            "language",
+            "locale",
+            "dynamic_resource",
+        ]
+        _dic = self.nlp_params()
```

Comment:
Its unclear why you are using the `nlp_params ` here. The `nlp_params ` method is only used for validation of certain params attributes. Why don't we just iterates through all the object attributes and return them in a dictionary?

Reply:
I see, I thought that the `nlp_params` is preferred method to retrieve nlp params, so I wanted to reuse that. I want to add this function to support serialization for both Params and FrozenParams object.